csharp/src/Microsoft.ML.OnnxRuntime/NamedOnnxValue.cs

```diff
@@ -218,12 +218,22 @@ out nativeTensor
                                                     ));
 
                     // fill the native tensor, using GetValue(index) from the Tensor<string>
-                    string[] stringsInTensor = new string[tensorValue.Length];
+                    var len = tensorValue.Length;
+                    var stringsInTensor = new IntPtr[len];
+                    var pinnedHandles = new GCHandle[len];
                     for (int i = 0; i < tensorValue.Length; i++)
                     {
-                        stringsInTensor[i] = tensorValue.GetValue(i);
+                        var utf8str = UTF8Encoding.UTF8.GetBytes(tensorValue.GetValue(i) + "\0");
```

Comment:
>UTF8Encoding.UTF8.GetByte [](start = 38, length = 25)

Where do I find the reverse conversion when the results come back from ORT?

Reply:
The reverse operation is done in NativeOnnxTensorMemory.cs. See below for the exact location.

https://github.com/microsoft/onnxruntime/blob/master/csharp/src/Microsoft.ML.OnnxRuntime/NativeOnnxTensorMemory.cs#L104

Reply:
I looked at the code and I believe the conversion is not handled properly that way either.

---
In reply to: [359645025](https://github.com/microsoft/onnxruntime/pull/2697#discussion_r359645025) [](ancestors = 359645025)